docs/src/integrity/integrity.rst

```diff
@@ -0,0 +1,60 @@
+***********************
+Computational Integrity
+***********************
+
+The integrity module of Opaque ensures that the untrusted job driver hosted on the cloud service schedules tasks in the manner computed by Spark's Catalyst query optimizer. 
+Opaque runs on Spark, which utilizes data partitioning to speed up computation. 
+Specifically, Catalyst will compute a physical query plan for a given dataframe query and delegate Spark workers (run on enclaves) to compute Spark SQL operations on data partitions. 
+Each of these individual units is trusted, but the intermediary steps in which the units communicate is controlled by the job driver, running as untrusted code in the cloud. 
+The integrity module will detect if the job driver has deviated from the query plan computed by Catalyst.
+
+Overview
+--------
+The main idea behind integrity support is to tag each step of computation with a MAC, attached by the enclave worker when it has completed its computation. 
```

Comment:
Can you elaborate on what exactly the MAC is over?

Reply:
See new commit

Reply:
Would be good to elaborate a bit further on what is MAC'd over -- maybe for example you can explain what is in each LogEntry, similar to what we say in the "Building Blocks" section of [this document](https://hackmd.io/IBokMEEEQh2Y0g_h9hw_NA) (but of course with the updated fields)

Reply:
Added the flatbuffers schema explicitly and a short description of each object in 2db9389